.github/workflows/test-and-publish.yaml

```diff
@@ -40,13 +40,18 @@ jobs:
         with:
           path: |
             ./_spidermonkey_install/*
-          key: spidermonkey102.13-${{ runner.os }}-${{ runner.arch }}
+          key: spidermonkey115.1.0-${{ runner.os }}-${{ runner.arch }}
           lookup-only: true # skip download
       - name: Setup Poetry
         if: ${{ steps.cache-spidermonkey.outputs.cache-hit != 'true' }}
         uses: snok/install-poetry@v1
         with:
           version: 1.5.1
+      - name: Setup XCode
+        if: ${{ matrix.os == 'macos-13' && steps.cache-spidermonkey.outputs.cache-hit != 'true' }}
+        # SpiderMonkey 115 ESR requires XCode SDK version at least 13.3
```

Comment:
Note this drops support for **compiling** on macOS version < 13 because the [Minimum OS Required for Xcode 14.3 with SDK 13.3 is macOS Ventura 13](https://developer.apple.com/support/xcode/).

See also:
* **https://firefox-source-docs.mozilla.org/widget/cocoa/sdks.html**
* https://hg.mozilla.org/releases/mozilla-esr115/file/f96bee7/build/moz.configure/toolchain.configure#l156
* https://bugzilla.mozilla.org/show_bug.cgi?id=1833998
* https://bugzilla.mozilla.org/show_bug.cgi?id=1773743
* https://bugzilla.mozilla.org/show_bug.cgi?id=1696513